Sources/MapboxNavigation/CarPlayNavigationViewController.swift

```diff
@@ -542,6 +553,33 @@ open class CarPlayNavigationViewController: UIViewController {
         }
     }
     
+    private func attemptToHighlightBuildings(_ progress: RouteProgress) {
```

Comment:
This is a copy from `NavigationViewController`. Even, fixme is copied. We need to extract this logic in a reusable algorithm.

Reply:
Agree, since there are also several class member properties involved in the building highlighting process I've created internal `BuildingHighlighting` protocol. Please check latest changes.